test/new-e2e/containers/k8s_test.go

```diff
@@ -0,0 +1,448 @@
+					suite.Equalf(stderr, "", "Standard error of `agent version` should be empty,")
+					match := versionExtractor.FindStringSubmatch(stdout)
+					if suite.Equalf(2, len(match), "'Commit' not found in the output of `agent version`.") {
+						suite.Equalf(GitCommit, match[1], "Agent isn’t running the expected version")
+					}
+				}
+			}
+		})
+	}
+}
+
+func (suite *k8sSuite) TestNginx() {
+	// suite.T().Parallel()
+
+	// `nginx` check is configured via AD annotation on pods
+	// Test it is properly scheduled
+	suite.testMetric("nginx.net.request_per_s",
+		[]string{},
+		[]*regexp.Regexp{
+			regexp.MustCompile(`^container_id:`),
+			regexp.MustCompile(`^container_name:nginx$`),
+			regexp.MustCompile(`^display_container_name:nginx`),
+			regexp.MustCompile(`^git\.commit\.sha:`),                                                       // org.opencontainers.image.revision docker image label
+			regexp.MustCompile(`^git\.repository_url:https://github\.com/DataDog/test-infra-definitions$`), // org.opencontainers.image.source   docker image label
+			regexp.MustCompile(`^image_id:ghcr\.io/datadog/apps-nginx-server@sha256:`),
+			regexp.MustCompile(`^image_name:ghcr\.io/datadog/apps-nginx-server$`),
+			regexp.MustCompile(`^image_tag:main$`),
+			regexp.MustCompile(`^kube_container_name:nginx$`),
+			regexp.MustCompile(`^kube_deployment:nginx$`),
+			regexp.MustCompile(`^kube_namespace:workload-nginx$`),
+			regexp.MustCompile(`^kube_ownerref_kind:replicaset$`),
+			regexp.MustCompile(`^kube_ownerref_name:nginx-[[:alnum:]]+$`),
+			regexp.MustCompile(`^kube_qos:Burstable$`),
+			regexp.MustCompile(`^kube_replica_set:nginx-[[:alnum:]]+$`),
+			regexp.MustCompile(`^kube_service:nginx$`),
+			regexp.MustCompile(`^pod_name:nginx-[[:alnum:]]+-[[:alnum:]]+$`),
+			regexp.MustCompile(`^pod_phase:running$`),
+			regexp.MustCompile(`^short_image:apps-nginx-server$`),
+		},
+	)
+
+	// `http_check` is configured via AD annotation on service
+	// Test it is properly scheduled
+	suite.testMetric("network.http.response_time",
+		[]string{},
+		[]*regexp.Regexp{
+			regexp.MustCompile(`^cluster_name:`),
+			regexp.MustCompile(`^instance:My_Nginx$`),
+			regexp.MustCompile(`^kube_cluster_name:`),
+			regexp.MustCompile(`^kube_namespace:workload-nginx$`),
+			regexp.MustCompile(`^kube_service:nginx$`),
+			regexp.MustCompile(`^url:http://`),
+		},
+	)
+
+	// Test KSM metrics for the nginx deployment
+	suite.testMetric("kubernetes_state.deployment.replicas_available",
+		[]string{
+			"kube_deployment:nginx",
+			"kube_namespace:workload-nginx",
+		},
+		[]*regexp.Regexp{
+			regexp.MustCompile(`^kube_cluster_name:`),
+			regexp.MustCompile(`^kube_deployment:nginx$`),
+			regexp.MustCompile(`^kube_namespace:workload-nginx$`),
+		},
+	)
+
+	// Check HPA is properly scaling up and down
+	// This indirectly tests the cluster-agent external metrics server
+	suite.testHPA("workload-nginx", "nginx")
+}
+
+func (suite *k8sSuite) TestRedis() {
+	// suite.T().Parallel()
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution. -->
**❓ question**
Why can't it run in // ?

Reply:
I initially wanted to run “everything” in //.
But it turns out that:
* running “classic” GO tests in // works (It’s what `T.Parallel()` is for !). But
* running suite subtests in // breaks everything. See [my tests here](https://gist.github.com/L3n41c/cc6d44abee4a90a90de0febf41351fb0).